src/Components/Components/src/Binding/CascadingModelBinder.cs

```diff
@@ -136,4 +133,54 @@ void IDisposable.Dispose()
     {
         Navigation.LocationChanged -= HandleLocationChanged;
     }
+
+    bool ICascadingValueComponent.CanSupplyValue(Type valueType, string? valueName)
+    {
+        var formName = string.IsNullOrEmpty(valueName) ?
+            (_bindingContext?.Name) :
+            ModelBindingContext.Combine(_bindingContext, valueName);
+
+        if (_bindingInfo != null &&
+            string.Equals(_bindingInfo.FormName, formName, StringComparison.Ordinal) &&
+            _bindingInfo.ValueType.Equals(valueType))
+        {
+            // We already bound the value, but some component might have been destroyed and
+            // re-created. If the type and name of the value that we bound are the same,
+            // we can provide the value that we bound.
+            return true;
+        }
+
+        // Can't supply the value if this context is for a form with a different name.
+        if (FormValueSupplier.CanBind(formName!, valueType))
+        {
+            var bindingSucceeded = FormValueSupplier.TryBind(formName!, valueType, out var boundValue);
+            _bindingInfo = new BindingInfo(formName, valueType, bindingSucceeded, boundValue);
+            if (!bindingSucceeded)
+            {
+                // Report errors
```

Comment:
What is this going to do?

Reply:
It will put errors in the binding context so that they can be consumed by the validator we'll plug in on the form.